### DISCUSSION TITLE
timestamps w/o time zone

### DISCUSSION THREAD

**Discussion by hmt** [porsager/postgres#329] [Q&A]
This is driving me nuts. I'm trying to set a date as timestamp coming from an input form:

I'm getting the date right from the DB (type timestamp w/o time zone): 

```js
let start = new Date(start_date).toISOString().replace('Z','');
```

Then I'm binding it to my input value (svelte code here, but you'll get how it works):
```html
   <input
	type="datetime-local"
	bind:value={start}
	on:change="{reset_start}">
```

So nothing spectacular here. Now setting this date via `sql`:
```js
await sql`UPDATE foo SET start = ${start} RETURNING *`;
```
It will for some reason add a time zone calculation and because I'm not CET the timestamp is off by two hours.

**Comment by porsager** [comment]
That's probably due to the time zone of your database. 

Try to post the output of running
```
SELECT current_setting('TIMEZONE');
```

You can set it with the connection options like this:
```
const sql = postgres({
  connection: {
    timezone: 'UTC'
  }
})
```

**Reply by PierBover** [reply]
Hey @porsager I'm wondering what will happen when setting the timezone in the connection options.

Will the client convert the dates into JS Date objects with a given timezone?

What will happen with `timestamp` and `timestampz` values?

**Comment by hmt** [comment]
Thank you, I did as advised:

```
export const sql = postgres({
  connection: {
    timezone: 'Europe/Berlin'
  }
});
```

And also have set the correct time zone in my db setting:
```
=> SELECT current_setting('TIMEZONE');
 current_setting
-----------------                                                                                                                               
 Europe/Berlin                                                                                                                                     
(1 row) 
```

There is still the two hour difference in between.

**Reply by porsager** [reply]
Well doesn't that makes sense since it's the utc time you're setting as local? 

Why are you stripping Z?

**Reply by hmt** [reply]
I'm stripping "Z" because the html input wouldn't read it otherwise.

Now I've set everything to UTC but that doesn't make a difference either. For some reason it still converts to local time.

**Reply by hmt** [reply]
is there a chance to just get the string value of the column?

Otherwise I really love the postgres module. Works really nice!

**Reply by porsager** [reply]
This issue doesn't seem related to Postgres.js but seems more about a svelte/browser issue 😋 Well. Let's see if we can fix it anyway

Try to post the actual value you set in the input value field, and what you'd expect to see.

**Reply by porsager** [reply]
Also you should almost never use timestamp  without time zone in Postgres

You remove Postgres ability to understand time zones, which could very well be causing your issue. 

Try to use timestamptz and don't do anything special anywhere else. Remove the toISOstring and z replace and pass dates directly.

**Reply by hmt** [reply]
sorry, don't want to ruin your Friday night...

This is what I'm setting:
```js
console.log(`UPDATE foo SET start = ${data.start} RETURNING *`)
res = await sql`UPDATE foo SET start = ${data.start} RETURNING *`;
```

The log looks fine:
```
UPDATE foo SET start = 2022-03-21T00:00 RETURNING *
```

In the browser (current FF) I'm also logging the value which is sent: `2022-03-21T00:00`
The returned value however is `start: "2022-03-20T22:00:00.000Z"`

Checking the value via psql:

```
> select start from foo; 
 start                                                                                                                           
---------------------                                                                                                                      
 2022-03-20 23:00:00                                                                                                            
(1 row)   
```

Now everything is mixed up. I think for a while I had at least the correct value via psql :S

**Reply by hmt** [reply]
ok, will change types to time zone now to see if that fixes it.

**Reply by porsager** [reply]
Haha. Don't worry - Time is a mess 🤣

**Comment by hmt** [comment]
ok, got it. I do need to strip the "Z" otherwise the browser ignores the date.
I have set my server to "Europe/Berlin" now with the type timestamp with time zone. Before I set the timestamp I'm adding the "Z" to the string again and I seem to be ok now. Reading the date with `date.toLocaleDateString()` I need to set it to `UTC`. I don't get it but it's ok, it does the job.

**Reply by porsager** [reply]
Won't svelte accept an actual js Date object?

**Reply by hmt** [reply]
Don't think it's svelte related: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local
Try adding the Z
